### ISSUE TITLE
No results for HTTP or Queue logs

### ISSUE THREAD

**Issue by dmason30** [laravel/vapor-ui#16]
- Vapor UI Version: 0.0.6
- Laravel Version: 8.1.0
- PHP Version: 7.4
- Database Driver & Version: MySQL 8.0

The below is the same on test/stage and production.

If I go to the Queue or HTTP logs I never get any results. 
The HTTP page polls for a while and then says no results found. 
The Queue page just seems to endlessly poll for new results but never gets any.

Is there a setting in AWS I need to enable to get output for these sections?

I get logs for CLI with the `schedule:run` output. 

![image](https://user-images.githubusercontent.com/20278756/93122282-23057200-f6be-11ea-80ae-a28eee4be2f7.png)

I also have results on the great new `Metrics` page. 
![image](https://user-images.githubusercontent.com/20278756/93122392-4b8d6c00-f6be-11ea-8483-2b60d2b19bc4.png)

## vapor.yml 
```yaml
environments:
  production:
    timeout: 900
    gateway-version: 2
    runtime: php-7.4
    cache: xxxxx
    cli-concurrency: 10
    cli-memory: 256
    cli-timeout: 900
    concurrency: 30
    database: xxxxx
    domain: xxxxx
    memory: 256
    storage: xxxxx
    queue-concurrency: 10
    queue-memory: 256
    warm: 5
```

**Comment by nunomaduro** [comment]
This can happen when they're non-relevant logs in CloudWatch, and really nothing to show you.

So, the current logic keeps searching for relevant logs until the current date.

Ideally, we would add something that prevents this from happening. Just like:

```
No logs were found until 2020-12-01. Keep searching?
```

And people can opt by keep searching until the current date.

**Comment by dmason30** [comment]
Sounds, like a good idea. Thanks.

I was looking at the AWS logs and it appears for both HTTP and Queue logs there is no useful output in them just the request start and end entries that you correctly filter out.

It would be great if we could get logs for the requests coming in similar to telescope and jobs in the queue logs. 

I just wandered if this was possible by enabling more detailed logs in AWS somewhere or would this have to be built in to vapor?

Also is there any plans to remove the CLI '/dev/null 2>&1' so that we can actually see the output of the commands?

**Comment by nunomaduro** [comment]
> I just wandered if this was possible by enabling more detailed logs in AWS somewhere for Lambda or would this have to be built in to vapor?

We would have to study it.

> Also is there any plans to remove the CLI /dev/null 2>&1 so that we can actually see the output of the commands in the log entries?

Can you point me to what are you talking about?

**Comment by dmason30** [comment]
@nunomaduro I was referring to the logs you see in the CLI output:

![image](https://user-images.githubusercontent.com/20278756/93237248-c1094300-f777-11ea-9c55-48f7bb5c9edf.png)

**Comment by jcorrego** [comment]
Hi, im also having some problem, but is due to the default timestamp used in all views...
<img width="1161" alt="Captura de Pantalla 2020-09-15 a la(s) 11 42 33 a  m" src="https://user-images.githubusercontent.com/2152949/93239559-92c03f00-f748-11ea-9996-5deccb4d41fb.png">

The default time, is always in the future (5 hours, because i am on UTC-5 timezone), so i have to manually substract 6 or seven hours to be able to se anything, and of it changes again all the time...

**Comment by nunomaduro** [comment]
@jcorrego The default time is the current time, in your timezone. If that is not happening, please create an issue for it.

**Comment by mbryne** [comment]
I would love to also view the requirement to view console output as outlined by https://github.com/laravel/vapor-ui/issues/16#issuecomment-692824284

I spent the day troubleshooting that last week. Because no logs are generated from the scheduled tasks emailOutputTo throws a SwiftTransport exception as there is no console output from those tasks.

The advice I received was to rebuild the scheduled tasks as queued jobs and use the Log facade. That makes sense but it would still be good to be able to view the output from scheduled tasks in the vapor logs.

**Comment by nunomaduro** [comment]
> I spent the day troubleshooting that last week. Because no logs are generated from the scheduled tasks emailOutputTo throws a SwiftTransport exception as there is no console output from those tasks.

@mbryne Normally those exceptions are logged by the exception handler. You should have those exceptions if you filter by "Error" or by "Exception".

**Comment by mbryne** [comment]
The issue is not about capturing the output of the exception handler. That's no troubles. It's capturing the console output of scheduled tasks to vapor ui logs / cloudwatch etc that I am personally keen on. 

As scheduled tasks pipe their output to null the scheduled tasks logs aren't available in vapor / cloudwatch. The output of command run via `php artisan` show up but scheduled tasks artisan calls are not.

As @dmason30 mentioned here https://github.com/laravel/vapor-ui/issues/16#issuecomment-692759986 this seems more like a question about scheduled tasks handling in the Laravel Framework proper than an issue with vapor-ui

More info and other examples are also mentioned here:  https://laracasts.com/discuss/channels/vapor/emailoutputonfailure-on-command-executing-every-time-in-vapor

**Comment by nunomaduro** [comment]
> The issue is not about capturing the output of the exception handler. That's no troubles. It's capturing the console output of scheduled tasks to vapor ui logs / cloudwatch etc that I am personally keen on.

Added a todo task internally, as it is not exactly related to vapor ui. Thanks!

**Comment by mbryne** [comment]
> > The issue is not about capturing the output of the exception handler. That's no troubles. It's capturing the console output of scheduled tasks to vapor ui logs / cloudwatch etc that I am personally keen on.
> 
> Added a todo task internally, as it is not exactly related to vapor ui. Thanks!

Much appreciated, happy to log a ticket elsewhere if it seems more appropriate.

**Comment by dmason30** [comment]
So I just deployed some changes that used `\Log::info` in a queued job in the hope that it would appear in the Vapor UI logs page. It doesn't, however, if I go to AWS CloudWatch `vapor-project-staging-queue` I can see the logs are recorded. Is there something I am missing or should Vapor UI be displaying these logs?

![image](https://user-images.githubusercontent.com/20278756/96277695-825fe600-0fcc-11eb-86f1-35ef5449ec53.png)

**Comment by nunomaduro** [comment]
If you are using the default queue, Vapor UI should have that logs.

**Comment by dmason30** [comment]
I have just noticed that my Log Streams are an hour behind but the actual log entries have the correct time as shown in previous comment. DST strikes again, I assume there isn't really any way to resolve this. 

<details>
<summary>Click to view log streams screen</summary>

![image](https://user-images.githubusercontent.com/20278756/96278580-90facd00-0fcd-11eb-9641-24678b9ea301.png)
</details>

If I change the Vapor logs filter to an hour previous to match the streams then I get results. The problem is that the filters only show up to 30 minutes old. I managed to get the filter to an hour before by manually editing the unix timestamp filter on the url.

<details>
<summary>Click to Vapor UI screenshot with results</summary>

![image](https://user-images.githubusercontent.com/20278756/96278819-de773a00-0fcd-11eb-9215-3d543f137cff.png)
</details>
